docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,112 @@
+For example, a node fails two uptime checks within a timespan of 1 minute. The second failure does not give us very much additional information does it? We already knew the node was offline less than one minute ago. If the difference were one hour that information might be more useful in determining uptime.
+
+We might be able to account for this by adjusting _w_ based on how much time has elapsed since last contact.
+To do this we'll need a new configurable `Uptime Scale`: a duration against which to scale the weight of an individual uptime check. To demonstrate, let's take an Uptime Scale of one hour. If a node is found to be offline when audited and its last contact is greater than or equal to one hour ago, the weight is scaled to 100%. This check will result in the full reputation punishment. A second failed check occurs 30 minutes later. This is one half the scale: the weight is reduced by 50%. Another failed check occurs one minute later, and the weight is scaled back to 1/60 the punishment.
+
+This implementation still exhibits the problem that, for nodes whose last contact was over one hour ago, we could concievably give them a full punishment even if they were only offline for one minute.
+
+## Implementation
+
+1. Remove old uptime reputation values from codebase:
+
+    uptime_success_count<br>
+    total_uptime_count<br>
+    uptime_reputation_alpha<br> 
+    uptime_reputation_beta<br>
+
+2. Add `downtime_suspended` and `under_review` timestamp columns to the nodes table and rename the `suspended` column to `audit_suspended`
+
+3. Implement new logic in estimation chore to suspend nodes
+    - Refactor downtimeTrackingDB method `GetOfflineTime` to check for entries which contain downtime from outside the measurement period
+
+        There is a particular characteristic of downtime tracking to take note of here:<br>
+        
+        A downtime entry tracked at March 2nd 00:00 contains downtime for some length of time preceding that point. See [downtime tracking](storage-node-downtime-tracking.md)<br>
+        
+        If we want to measure total downtime starting at March 1st 00:00, and there is a March 1st 01:00 entry indicating 2 hours of downtime, we need to truncate the measured downtime in our calculation in order to avoid unfairly including downtime which occurred outside the tracking period.<br>
+        
+        We should be able to solve this problem by taking the first entry within the tracking period and checking if the offline time it contains extends beyond the start of the tracking period. If so, truncate it to the start of the tracking period, then sum the rest of the entries as usual.
+
+    - Refactor relevant overlay methods to handle downtime_suspended nodes: `KnownReliable`, `FindNodes`, etc.
```

Comment:
This is a little off topic, but I've been wondering for a while if we should multiple different tables or possibly a node_type column that summarizes some of the logic of seeing if a node is new, reliable, DQ'ed, etc.

Reply:
Yeah, forget I mentioned this... its different design doc that's only partially formed in my brain.  I was looking into how to get random samples from SQL, and having an accurate count of records can help.  It turns out having an accurate count of records is its own problem which basically trades slightly faster select speed for much slower insert speed... so its probably not worth it at this point in the game.